### PR TITLE
Reactive Receptor

### PR DESCRIPTION
**Pull request by pawelkaczor**

### CODE REVIEW

**Comment by pawelkaczor** [comment]
@ktoso could you please review this PR? The goal (which I believe has been achieved :) ) was to introduce back-pressure to the following event flow: Business Process Stream -> Receptor. Role of the Receptor is to forward received events ensuring ALOD.

Big picture: https://github.com/pawelkaczor/akka-ddd/wiki/Saga

Thanks in advance.

**Comment by ktoso** [comment]
I'll give it a look - hopping on a long haul flight now, not sure if I'll have internet though.

**Comment by pawelkaczor** [comment]
Ok, great, have a nice flight!

File: akka-ddd-core/src/main/scala/pl/newicom/dddd/process/Receptor.scala [hunk 1]
```diff
@@ -72,20 +73,29 @@ abstract class Receptor extends AtLeastOnceDeliverySupport {
 
   override lazy val persistenceId: String = s"Receptor-${config.stimuliSource.officeName}-${self.path.hashCode}"
 
+  var inFlightCallback: InFlightMessagesCallback = null
```
Comment (ktoso) [inline akka-ddd-core/src/main/scala/pl/newicom/dddd/process/Receptor.scala]: Option? ;)

**Pull request merged**